src/pages/scientificServices/pipelines/views/JobHistory.tsx

```diff
@@ -33,16 +33,30 @@ export const JobHistory = () => {
   const [pageNumber, setPageNumber] = useState(1);
   const [itemsPerPage, setItemsPerPage] = useState(10);
   const [pipelineRunsResponse, setPipelineRunsResponse] = useState<GetPipelineRunsResponse>();
-  const nextPageToken = useRef<string>();
+  const [isLoading, setIsLoading] = useState(false);
+  const [sort, setSort] = useState<{ field: string; direction: 'asc' | 'desc' }>({
```

Comment:
as a typescript noob, what is happening here?

Reply:
this is an inline way of defining the expected type for the `sort` state.

`direction: 'asc' | 'desc'` is the most interesting bit, which enforces that not only is the`direction` property defined for the object, but it must be either "asc" or "desc"


but per your related question about this below, i've extracted this into its own interface since that's a cleaner way of doing it and how i should have done it from the start:

```
interface SortProperties {
  field: string;
  direction: 'asc' | 'desc';
}
```

committed in 6e786599d59b6806899d1a3d6c429a9989319e83

Reply:
thanks for the explanation